glotaran/analysis/scheme.py

```diff
@@ -177,11 +178,9 @@ def prepared_data(self) -> typing.Dict[str, xr.Dataset]:
             if isinstance(dataset, xr.DataArray):
                 dataset = dataset.to_dataset(name="data")
 
-            if "weight" in dataset and "weighted_data" not in dataset:
-                dataset["weighted_data"] = np.multiply(dataset.data, dataset.weight)
+            self._add_weight(label, dataset)
```

Comment:
This seems to be the crucial place where the weights are added, but the function prepared_data is only every called in the tests (test_scheme.py). So for a normal analysis no weights are added.